proxy/http2/Http2ConnectionState.cc

```diff
@@ -1355,16 +1353,10 @@ Http2ConnectionState::delete_stream(Http2Stream *stream)
 }
 
 void
-Http2ConnectionState::release_stream(Http2Stream *stream)
+Http2ConnectionState::release_stream()
```

Comment:
I know this is called `release_stream` by historical reason. It might be time to rename because this is a completely connection specific function now.

Reply:
Yes, I completely agree.  Leaving that for @a-canary to address in his refactoring.